LobbyServer2/BridgeServer/Game.cs

```diff
@@ -558,48 +560,202 @@ protected bool FillTeam(List<long> players, Team team, GameSubType gameSubType)
+        }
+        
+
         for (int i = 0; i < botNum; i++)
         {
-            LobbyServerPlayerInfo playerInfo = AddBot(team);
+            LobbyServerPlayerInfo playerInfo = AddBot(team, i, gameSubType);
             log.Info($"adding bot {playerInfo.CharacterType} to {team}");
         }
 
         return true;
     }
 
-    protected LobbyServerPlayerInfo AddBot(Team team)
+    protected LobbyServerPlayerInfo AddBot(Team team, int botNr, GameSubType gameSubType)
     {
-        CharacterType characterType = PickCharacter(team, true);
+        // Initialize basic character information
+        CharacterType characterType = CharacterType.None;
+        LobbyCharacterInfo lobbyCharacterInfo = new();
+        LobbyServerPlayerInfo controllingPlayer = new();
+        bool isAntiSocial = gameSubType.Mods is not null && gameSubType.Mods.Contains(GameSubType.SubTypeMods.AntiSocial);
+        long accountId = 0;
+
+        // Check if FourLancher mode is enabled
+        if (IsFourLancher)
+        {
+            // Handle AntiSocial games
+            if (isAntiSocial)
+            {
+                if (team == Team.TeamA)
+                {
+                    // Team A: has controlling player
+                    controllingPlayer = GetControllingPlayer(team);
+                    PersistedAccountData account = DB.Get().AccountDao.GetAccount(controllingPlayer.AccountId);
+                    accountId = controllingPlayer.AccountId;
+                    if (account?.AccountComponent?.LastRemoteCharacters != null &&
+                        botNr >= 0 && botNr < account.AccountComponent.LastRemoteCharacters.Count)
+                    {
+                        characterType = account.AccountComponent.LastRemoteCharacters[botNr];
+                        CharacterComponent characterComponent = (CharacterComponent)account.CharacterData[characterType].CharacterComponent.Clone();
+                        lobbyCharacterInfo = LobbyCharacterInfo.Of(account.CharacterData[characterType], characterComponent);
+                    }
+                    else
+                    {
+                        // Fallback if LastRemoteCharacters is null or botNr is out of range pick a random one for that player and let them controll it
+                        characterType = PickCharacter(team, true);
+                        CharacterComponent characterComponent = (CharacterComponent)account.CharacterData[characterType].CharacterComponent.Clone();
+                        lobbyCharacterInfo = LobbyCharacterInfo.Of(account.CharacterData[characterType], characterComponent);
+                    }
+                }
+                else if (team == Team.TeamB)
+                {
+                    // Team B: Uncontrolled bots (no controlling player)
+                    characterType = PickCharacter(team, true);
+                    lobbyCharacterInfo = InitializeDefaultCharacterInfo(characterType);
+                }
```

Comment:
So this part is for `IsAntiSocial && team == Team.TeamB` and the code for otherwise is duplicated?